addons/mail/models/mail_channel_partner.py

```diff
@@ -45,3 +45,15 @@ def write(self, vals):
             if {'channel_id', 'partner_id', 'partner_email'} & set(vals):
                 raise AccessError(_('You can not write on this field'))
         return super(ChannelPartner, self).write(vals)
+
+    def mail_channel_partner_format(self, partner_infos=None):
+        members = self.read(['id', 'partner_id'])
+        if partner_infos and len(partner_infos):
+            formatted_members = []
+            for member in members:
+                formatted_members.append({
+                    'id': member['id'],
+                    'partner': partner_infos[member['partner_id'][0]],
```

Comment:
Are you sure that each `member -> partner` is part of `partner_infos`? I feel like it might not be the case for some of them, depending on how it is filled originally.